app/team.go

```diff
@@ -592,12 +592,7 @@ func (a *App) GetTeam(teamId string) (*model.Team, *model.AppError) {
 }
 
 func (a *App) GetTeamByName(name string) (*model.Team, *model.AppError) {
-	result := <-a.Srv.Store.Team().GetByName(name)
-	if result.Err != nil {
-		result.Err.StatusCode = http.StatusNotFound
-		return nil, result.Err
-	}
-	return result.Data.(*model.Team), nil
+	return a.Srv.Store.Team().GetByName(name)
```

Comment:
This is not exactly the same, here we are changing the StatusCode of the returned error to `StatusNotFound` which is not the underlaying returned error (StatusInternalServerError).